inference-engine/src/mkldnn_plugin/nodes/mkldnn_reorder_node.cpp

```diff
@@ -161,14 +172,48 @@ bool MKLDNNReorderNode::created() const {
     return getType() == Reorder;
 }
 
+void MKLDNNReorderNode::optimizedNspc2Ncsp() {
+    auto parentEdge = getParentEdgeAt(0);
+    auto childEdge = getChildEdgeAt(0);
+    const int ndims = parentEdge->getDims().ndims();
+    const size_t DIM0 = parentEdge->getDims()[0];
+    const size_t DIM1 = parentEdge->getDims()[1];
+    const size_t DIM2 = ndims == 5 ? parentEdge->getDims()[ndims - 3] : 1;
+    const size_t DIM3 = parentEdge->getDims()[ndims - 2];
+    const size_t DIM4 = parentEdge->getDims()[ndims - 1];
+
+    auto src_data = reinterpret_cast<const float *>(parentEdge->getMemoryPtr()->GetPtr());
+    auto dst_data = reinterpret_cast<float *>(childEdge->getMemoryPtr()->GetPtr());
+
+    parallel_for4d(DIM0, DIM2, DIM3, DIM4, [&](size_t dim0, size_t dim2, size_t dim3, size_t dim4) {
+        for (size_t dim1 = 0; dim1 < DIM1; dim1++) {
+            size_t src_off = dim0 * DIM2 * DIM3 * DIM4 * DIM1 +
+                             dim2 * DIM3 * DIM4 * DIM1 +
+                             dim3 * DIM4 * DIM1 +
+                             dim4 * DIM1 +
+                             dim1;
+            size_t dst_off = dim0 * DIM1 * DIM2 * DIM3 * DIM4 +
+                             dim1 * DIM2 * DIM3 * DIM4 +
+                             dim2 * DIM3 * DIM4 +
+                             dim3 * DIM4 +
+                             dim4;
```

Comment:
I am not sure if extract calculation with constant out of loop would give higher performance in this case. such as get: block_ncsp[0] = DIM1 * DIM2 * DIM3 * DIM4, 
block_ncsp[1] = DIM2 * DIM3 * DIM4
block_ncsp[2] = DIM3 * DIM4
, then use block_ncsp directly in each iteration. Deserve to try?